### PR TITLE
fix: L2 improvements

### PR DESCRIPTION
**Pull request by JChiaramonte7**
### Changes

- [x] Redirect if on shorting and on L2
- [x] Clear selected synths in exchange on network switch
- [x] Link orders in notification window to correct block explorer
- [x] Expand decimals places on synth balances to 4
- [x] Redirect user to staking when synth balances are empty on L2

fixes https://github.com/Kwenta/issues/issues/41

### CODE REVIEW

File: sections/exchange/hooks/useExchange.tsx [hunk 1]
```diff
@@ -559,6 +561,14 @@ const useExchange = ({
 		return null;
 	}, [feeAmountInBaseCurrency, basePriceRate]);
 
+	useEffect(() => {
+		setCurrencyPair({
+			base: null,
+			quote: null,
+		});
+		// eslint-disable-next-line	react-hooks/exhaustive-deps
+	}, [network.id]);
```
Comment (JChiaramonte7) [inline sections/exchange/hooks/useExchange.tsx:570]: it wants me to add setCurrencyPair to the deps but this causes this `useEffect` to run excessively and eventually crashes the page with too many queries to local storage 

(we store selected synths in local storage to persist state between refreshes)

Comment (kmeraz) [reply sections/exchange/hooks/useExchange.tsx:570]: can we cache `setCurrencyPair` via `useCallback` and appease the linter? does that work?

Comment (JChiaramonte7) [reply sections/exchange/hooks/useExchange.tsx]: ended up fixing the issue upstream 

https://github.com/Synthetixio/kwenta/pull/469/files#diff-c2167a32a8bb0c5741361d7e5ea6df1926f47c932438f401c5fb46bd2b9049f3R7

File: pages/shorting/index.tsx [hunk 2]
```diff
@@ -16,11 +16,23 @@ import { PageContent, MainContent, RightSideContent, FullHeightContainer } from
 import { DesktopOnlyView } from 'components/Media';
 
 import { isWalletConnectedState, isL2State } from 'store/wallet';
+import { useRouter } from 'next/router';
+import ROUTES from 'constants/routes';
+import { useEffect } from 'react';
```
Comment (kmeraz) [inline pages/shorting/index.tsx:21]: nit: import useEffect up top on line 1

**Review by kmeraz — APPROVED** [review]

**Pull request merged**